pkg/operator/reporting/templates_test.go

```diff
@@ -0,0 +1,569 @@
+			reportTemplate: &ReportQueryTemplateContext{
+				ReportQuery:       newTestReportQuery(testReportName, testNamespace, ds1, "SELECT * FROM {| dataSourceTableName .Report.Inputs.UnreadyDeploymentReplicas |}"),
+				ReportDataSources: nil,
+				PrestoTables: []*metering.PrestoTable{
+					newTestPrestoTable("", testNamespace, testNamespace, testCatalogName, nil),
+				},
+			},
+			templateContext: TemplateContext{
+				Report: ReportTemplateInfo{
+					Inputs: map[string]interface{}{"UnreadyDeploymentReplicas": ""},
+				},
+			},
+			expectErr:    true,
+			expectErrMsg: fmt.Sprintf("error executing template: template: reportQueryTemplate:1:17: executing \"reportQueryTemplate\" at <dataSourceTableName ...>: error calling dataSourceTableName: ReportDataSource %s dependency not found", ""),
+		},
+		{
+			name: "ReportDataSource dependencies were not found (due to Report.Inputs referencing a ReportQuery) results in error",
+			reportTemplate: &ReportQueryTemplateContext{
+				ReportQuery:       newTestReportQuery(testReportName, testNamespace, ds1, "SELECT * FROM {| dataSourceTableName .Report.Inputs.UnreadyDeploymentReplicas |}"),
+				ReportDataSources: []*metering.ReportDataSource{ds1},
+				PrestoTables: []*metering.PrestoTable{
+					newTestPrestoTable(ds1.Status.TableRef.Name, testNamespace, testSchemaName, testCatalogName, nil),
+				},
+			},
+			templateContext: TemplateContext{
+				Report: ReportTemplateInfo{
+					Inputs: map[string]interface{}{"UnreadyDeploymentReplicas": testReportQuery},
+				},
+			},
+			expectErr:    true,
+			expectErrMsg: fmt.Sprintf("error executing template: template: reportQueryTemplate:1:17: executing \"reportQueryTemplate\" at <dataSourceTableName ...>: error calling dataSourceTableName: ReportDataSource %s dependency not found", testReportQuery),
+		},
+		{
+			name: "ReportDataSource dependencies were not found (due to Report.Inputs referencing an integer) results in error",
+			reportTemplate: &ReportQueryTemplateContext{
+				ReportQuery:       newTestReportQuery(testReportName, testNamespace, ds1, "SELECT * FROM {| dataSourceTableName .Report.Inputs.UnreadyDeploymentReplicas |}"),
+				ReportDataSources: []*metering.ReportDataSource{ds1},
+				PrestoTables: []*metering.PrestoTable{
+					newTestPrestoTable(ds1.Status.TableRef.Name, testNamespace, testSchemaName, testCatalogName, nil),
+				},
+			},
+			templateContext: TemplateContext{
+				Report: ReportTemplateInfo{
+					Inputs: map[string]interface{}{"UnreadyDeploymentReplicas": 5},
+				},
+			},
+			expectErr:    true,
+			expectErrMsg: "error executing template: template: reportQueryTemplate:1:44: executing \"reportQueryTemplate\" at <.Report.Inputs.Unrea...>: wrong type for value; expected string; got int",
+		},
+		{
+			name: "ReportDataSource dependencies were not found (due to Report.Inputs referencing a *time.Time object) results in error",
```

Comment:
Same here, I was mostly thinking of having a new input definition expecting a `type: time` with a field name that isn't reportingStart/reportingEnd (those are special and are assumed to be time) and testing providing it/not providing it/etc.